### PR TITLE
Add type annotations

### LINKED ISSUE #26: Add support for typing annotations
It would be nice if OrderedSet supported PEP 484 typing module, so that we can do in Python 3:
```
my_set: OrderedSet[Context] = OrderedSet()
```

Right now pycharm/pylint flag it as a typing error:
```
Class 'type' does not define '__getitem__', so the '[]' operator cannot be used on its instances
```
Related: PyCQA/pylint#1657

### PR DESCRIPTION
**Pull request by rominf**
Fix #26

### CODE REVIEW

File: ordered_set.pyi [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+T = TypeVar('T')
+
+
+class OrderedSet(MutableSet[T], Sequence[T]):
+    def __init__(self, iterable: Iterable[T] = None) -> None:
+        self.map: Dict[T, int] = {}
+        self.items: List[T] = []
+
+    def __len__(self) -> int: ...
+    def __getitem__(self, index: Union[int, slice, Sequence[T]]) -> OrderedSet[T]: ...
```
Comment (rspeer) [inline ordered_set.pyi]: Shouldn't the signature of fancy indexing be `Sequence[int]` instead of `Sequence[T]`?

Comment (rominf) [reply ordered_set.pyi]: Agreed. My mistake. Fixed it.

**Comment by rominf** [comment]
I think the only way to write tests for type annotations is to use https://github.com/agronholm/typeguard. But the project is not mature enough in my opinion.

**Comment by rspeer** [comment]
Okay.

It seems that Travis stopped supporting their `3.7-dev` Python environment today, and the `3.7` replacement can't be tested at the same time as 3.4. I've removed testing on 3.7 so that Travis tests can pass. If you merge master into your branch, it should pass the checks.

**Comment by rominf** [comment]
@rspeer, can you release a version with type hinting on PyPI?

**Comment by rspeer** [comment]
Done.

**Comment by rominf** [comment]
Thank you!

**Pull request merged**